kotlin/workflow-runtime/src/main/java/com/squareup/workflow/internal/WorkflowNode.kt

```diff
@@ -54,8 +59,13 @@ internal class WorkflowNode<PropsT, StateT, OutputT : Any, RenderingT>(
    * true after the worker has finished and we've reported that fact to the workflow. This is to
    * prevent the workflow from entering an infinite loop of getting `Finished` events if it
    * continues to listen to the worker after it finishes.
+   *
+   * @param debugIdReservation Empty object that is used to identify the worker session for
+   * [WorkflowDiagnosticListener]s. The object's hashcode is used as the ID, and the object is help onto
```

Comment:
s/help/held/

Reply:
Relying on uniqueness of hash codes seems sketchy. UUID?

Reply:
Using a simple `long` counter for both workflows and workers.